### ISSUE TITLE
do_download API could be much cleaner.

### ISSUE THREAD

**Issue by petersilva** [MetPX/sarracenia#74]
currently, the work a do_download script must do is:
msg = parent.msg
download msg.url -> msg.new_dir / msg.new_file   (implements inflight?)
then implement local attributes...
  -- chown, chmod, utime.

Every plugin must interpret at least half a dozen options for these settings (preserve*, inflight*) etc...
There is no variabilty in setting the cwd, selection of the inflight file name, nor in setting attributes, so those should not be part of the do_download responsibilities.  It would be better to

cd msg.new_dir
    do download is only responsible to download msg.url -> msg.new_inflight_file ?
and then built-in code takes care of all the attributes, etc...

This is already half implemented in the sr_util/sr_transport class.  Essentially what do_download should be doing is identical to the *get* function of sr_transport.

Changes would be:
- change new_lock -> msg.new_inflight_file.
- change the name of *get* to *do_download* in all sr_transport, sr_http, sr_ftp.
- figure out how to call declare each declared do_download as an sr_transport.

perhaps derive a plugin_transport as an sr_transport?

anyways, would be much cleaner if it worked that way.

no incompatibility, just future scripts would have much less in them to be correct.

**Comment by petersilva** [comment]
perhaps this is what Michel was working towards with his sr_transport class.  I think he tried to explain it to me, but I didn't get it at the time. I guess the only objection, is that if we change the name of *get* to *do_download*, then we are done.

could just declare an sr_transport class object, and have it's *get* call the plugged in *do_download*.

**Comment by petersilva** [comment]
just noticed that do_get has been added... and I'm not sure what its interface is.

**Comment by michelgrenier** [comment]
yup... sr_transport is the place way down where most download/send options are taken into account.
except for local file thing which I did not have time to turn into a class.
